utils/check_modules.sh

```diff
@@ -36,40 +36,36 @@
 # (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
 # THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-set -e
-
 if [ ! -d "utils/sl" ];then
   cd ..
 fi
+
+/bin/rm -f pylint.log
+
+./utils/sl/check_script.py -s
+
+# Test the build code with daos_build.
+PYTHONPATH=$PWD/utils
+export PYTHONPATH
+
+./utils/sl/check_script.py -w SConstruct
+
+if [ -n "$CHANGE_TARGET" ]; then
+    ./utils/sl/check_script.py -w $(git diff --name-only origin/"$CHANGE_TARGET"... -- "*/SConscript")
+else
+    ./utils/sl/check_script.py -w $(git ls-files -i -x SConscript)
+fi
+
 # Set PYTHONPATH for source files not installed files
 PYTHONPATH=$PWD/utils:$PWD/src/tests/ftest/util/
 PYTHONPATH=$PYTHONPATH:$PWD/src/tests/ftest/cart/util/
 PYTHONPATH=$PYTHONPATH:$PWD/src/tests/ftest/util/apricot/
 PYTHONPATH=$PYTHONPATH:$PWD/src/client/
 export PYTHONPATH
 
-if [ -z "$*" ]; then
-  flist="-c utils/daos_build.py -s SConstruct"
-  # Exclude raft and utils/sl
-  scripts=$(find . -name SConscript | grep -v -e utils/sl -e raft \
-          -e build/external | sort)
-  for file in $scripts; do
-    flist+=" -s $file "
-  done
-  # the functional test code
-  flist+=" $(find src/tests/ftest/ -name \*.py | sort)"
-  flist+=" $(find src/client/pydaos/ -name \*.py | sort)"
-  flist+=" $(find src/client/dfuse/test/ -name \*.py | sort)"
-  flist+=" $(find src/cart/ -name \*.py | sort)"
-  flist+=" $(find utils/ -name \*.py | sort)"
+if [ -n "$CHANGE_TARGET" ]; then
+    ./utils/sl/check_script.py -x utils/sl $(git diff --name-only origin/"$CHANGE_TARGET"... -- "*.py")
```

Comment:
(style)  line over 80 characters
(lint) Quote this to prevent word splitting. [SC2046]